src/Raven.Server/Smuggler/Documents/DatabaseDestination.cs

```diff
@@ -543,145 +545,131 @@ private async ValueTask FinishBatchOfDocumentsAsync()
                 {
-                    return this;
-                }
-            }
-
+                    var parsedCommands = _commandParsedCommands.GetArraySegment();
 
-            private async ValueTask SendCommandsAsync(JsonOperationContext context)
-            {
-                if (_compareExchangeAddOrUpdateCommands.Count > 0)
-                {
-                    var compareExchangeAddOrUpdateCommands = _compareExchangeAddOrUpdateCommands;
-                    var (index, _) = await _database.ServerStore.SendToLeaderAsync(new AddOrUpdateCompareExchangeBatchCommand(_compareExchangeAddOrUpdateCommands, context, RaftIdGenerator.DontCareId));
-                    foreach (var command in compareExchangeAddOrUpdateCommands)
-                    {
-                        command.Value.Dispose();
-                    }
-                    var updateClusterTransactionIndex = new UpdateClusterTransactionIndex
-                    {
-                        Index = index
-                    };
-                    foreach (AddOrUpdateCompareExchangeCommand cmd in _compareExchangeAddOrUpdateCommands)
-                    {
-                        if(ClusterTransactionCommand.IsAtomicGuardKey(cmd.Key, out var docId) == false)
-                            continue;
-                        
-                        updateClusterTransactionIndex.DocumentIds.Add(docId);
-                    }
-                    _compareExchangeAddOrUpdateCommands.Clear();
+                    var raftRequestId = RaftIdGenerator.NewId();
+                    var options = new ClusterTransactionCommand.ClusterTransactionOptions(taskId);
+                    var topology = _database.ServerStore.LoadDatabaseTopology(_database.Name);
 
-                    if (updateClusterTransactionIndex.DocumentIds.Count > 0)
+                    var clusterTransactionCommand = new ClusterTransactionCommand(_database.Name, _database.IdentityPartsSeparator, topology, parsedCommands, options, raftRequestId);
+                    var (ctcIndex, _) = await _database.ServerStore.SendToLeaderAsync(clusterTransactionCommand);
+                    for (int i = 0; i < _commandParsedCommands.Length; i++)
                     {
-                        await _database.TxMerger.Enqueue(updateClusterTransactionIndex);
-                        foreach (var docId in updateClusterTransactionIndex.CompareExchangeToRemove)
-                        {
-                            _compareExchangeRemoveCommands.Add(new RemoveCompareExchangeCommand(_database.Name,
-                                ClusterTransactionCommand.GetAtomicGuardKey(docId),0, context, RaftIdGenerator.DontCareId, fromBackup: true));
-                        }    
+                        _commandParsedCommands[i].Document.Dispose();
                     }
+                    _commandParsedCommands.Clear();
+                    await _database.ServerStore.Cluster.WaitForIndexNotification(ctcIndex);
```

Comment:
I wonder if we can `await` for this at the end of the import.
Also, maybe is worth `await` for changes to be applied on the current node (`_database.ClusterTransactionWaiter.WaitForResults(taskId, default);`).

Reply:
I think we should do `_database.ClusterTransactionWaiter.WaitForResults(taskId, import-token)`